### DISCUSSION TITLE
[Question] Multiple Facebook/Meta Apps for Authentication depending on user type

### DISCUSSION THREAD

**Discussion by pnmcosta** [pocketbase/pocketbase#2234] [Q&A] [answered]
I'm evaluating Pocketbase to use for a new project MVP. Our use cases include 2 types of users, business and customers, due to the way Facebook Apps are implemented (permissions) I will need to leverage 2 apps for authentication.

I would be creating 2 distinct PB collections for these entities, Account and Customer.

I noticed there's a new Meta auth provider, how does this defer from Facebooks?

Would I be able to leverage Meta's for my business users and Facebook for customers?

Many thanks for any inputs.

**Comment by pnmcosta** [comment]
Thank you @ganigeorgiev I did not notice the Discussion tab before 🤦

**Comment by ganigeorgiev** [comment, marked as answer]
> I noticed there's a new Meta auth provider, how does this defer from Facebooks?
> Would I be able to leverage Meta's for my business users and Facebook for customers?

I'm not sure that I understand what do you refer by Meta provider. There is only one [Facebook OAuth2 client](https://github.com/pocketbase/pocketbase/blob/master/tools/auth/facebook.go) implementation at the moment.

If you want to have multiple Facebook OAuth2 apps you can try to use the generic OpenID Connect client (aka. `oidc`). Currently we allow up to 3 `oidc` clients (`oidc`, `oidc2` and `oidc3`).
I haven't personally tested Facebook with OIDC, but at least based on https://www.facebook.com/.well-known/openid-configuration/ it seems that it is supported.

**Reply by pnmcosta** [reply]
Apologies, I must have misunderstood the purpose of https://github.com/pocketbase/pocketbase/blob/master/models/settings/settings.go#L26 I thought this was related to auth provider, but clearly not. Could you clarify what this is for?

Got it, perhaps for the simpler auth flow (customers) I could use one of the oidc providers, many thanks for pointing these out to me.

**Reply by ganigeorgiev** [reply]
`Settings.Meta` is currently used to store some "meta" app configurations (eg. the app name, whether to hide from the UI the collection modify controls, etc.).
